python-package/lightgbm/basic.py

```diff
@@ -2336,13 +2336,13 @@ def _free_buffer(self):
         self.__is_predicted_cur_iter = []
         return self
 
-    def set_network(self, machines, local_listen_port=12400,
-                    listen_time_out=120, num_machines=1):
+    def set_network(self, machines: str, local_listen_port: int = 12400,
+                    listen_time_out: int = 120, num_machines: int = 1) -> "Booster":
         """Set the network configuration.
 
         Parameters
         ----------
-        machines : list, set or string
```

Comment:
I think adding support for behavior from old docstring will be better.
https://github.com/microsoft/LightGBM/blob/bcf443b568052eac05cb8b11b17be9926b862f8f/python-package/lightgbm/basic.py#L2203-L2208

Reply:
alright no problem

Reply:
Thanks!